browser/resources/settings/email_aliases_page/tests/email_aliases_modal.test.tsx

```diff
@@ -0,0 +1,270 @@
+        aliasCount={0}
+        onReturnToMain={mockOnReturnToMain}
+        emailAliasesService={mockEmailAliasesService}
+      />)
+    })
+
+    // Wait for generated alias
+    await waitFor(() => {
+      expect(mockEmailAliasesService.generateAlias).toHaveBeenCalled()
+    })
+
+    // Wait for loading to complete and alias to be generated
+    await waitFor(() => {
+      expect(screen.getByPlaceholderText(localeRegex(
+        'emailAliasesEditNotePlaceholder'))).toBeInTheDocument()
+      expect(screen.queryByTestId('loading-icon')).not.toBeInTheDocument()
+      const generatedEmailContainer = screen.getByText(localeRegex(
+        'emailAliasesAliasLabel')).closest('div')?.nextElementSibling
+      expect(generatedEmailContainer).toHaveTextContent('generated@brave.com')
+    })
+
+    // Ensure the save button is enabled
+    const saveButton = screen.getByText(localeRegex(
+      'emailAliasesCreateAliasButton'))
+    expect(saveButton).toHaveAttribute('isdisabled', 'false')
+
+    // Click save button
+    clickLeoButton(saveButton)
+
+    // Check that updateAlias was called
+    await waitFor(() => {
+      expect(mockEmailAliasesService.updateAlias).toHaveBeenCalled()
+      expect(mockOnReturnToMain).toHaveBeenCalled()
+    })
+  })
+
+  it('handles alias regeneration', async () => {
+    render(
+      <EmailAliasModal
+        viewState={{ mode: 'Create' }}
+        mainEmail={mockEmail}
+        aliasCount={0}
+        onReturnToMain={mockOnReturnToMain}
+        emailAliasesService={mockEmailAliasesService}
+      />
+    )
+
+    // Wait for initial generation
+    await waitFor(() => {
+      expect(mockEmailAliasesService.generateAlias).toHaveBeenCalled()
+      expect(screen.getByTitle('emailAliasesRefreshButtonTitle'))
+        .toBeInTheDocument()
+    })
+
+    // Click regenerate button
+    const regenerateButton = screen.getByTitle('emailAliasesRefreshButtonTitle')
+    await act(async () => {
+      clickLeoButton(regenerateButton)
+    })
+
+    // Check that generateAlias was called again
+    await waitFor(() => {
+      expect(mockEmailAliasesService.generateAlias).toHaveBeenCalledTimes(2)
+    })
+  })
+
+  it('shows limit reached message in bubble mode', async () => {
+    render(
+      <EmailAliasModal
+        viewState={{ mode: 'Create' }}
+        mainEmail={mockEmail}
+        aliasCount={5}
+        onReturnToMain={mockOnReturnToMain}
+        emailAliasesService={mockEmailAliasesService}
+        bubble={true}
+      />
+    )
+
+    // Wait for limit check
+    await waitFor(() => {
+      expect(screen.getByText(localeRegex('emailAliasesBubbleLimitReached')))
+        .toBeInTheDocument()
+    })
+  })
+
+  it('shows loading state while generating alias', async () => {
+    // Mock generateAlias to take some time
+    mockEmailAliasesService.generateAlias = jest.fn().mockImplementation(() => {
+      return new Promise(
+        resolve => setTimeout(() => resolve('new@brave.com'), 100))
```

Comment:
same as the other timeout, this makes the tests run longer and can lead to flaky tests

Reply:
I think resolving promises should already run in a microtask, so you should be fine to just have


```suggestion
        resolve => resolve('new@brave.com'))
```

here - it will still run asynchronously.

Reply:
yes, if the goal here is just to make it async then the promise should be sufficient

Reply:
fixed